### DISCUSSION TITLE
Don't require accessibility permissions to open "quick terminal"

### DISCUSSION THREAD

**Discussion by lishaduck** [ghostty-org/ghostty#3813] [Feature Requests, Ideas]
Uhhh... Warp doesn't require accessibility permissions for a "dedicated hotkey window", but Ghostty requires them for the "quick terminal." Getting IT to approve of granting apps accessibility permissions is difficult (though not impossible), so I'd like it if Ghostty was able to offer a (potentially pared-down) version that didn't require accessibility permissions.

**Comment by 00-kat** [comment]
I'm pretty sure that it's using a global keybinding (i.e. `keybind = global:foo=bar`) that's resulting in needing accessibility permissions, and not specifically for opening a quick terminal.

**Reply by lishaduck** [reply]
Yeah, the point is that it isn't needed for Warp, so something's different in there (just checked, and it does indeed work globally, not just when the warp window is in focus).

**Reply by mitchellh** [reply]
I don't know of a way to get global key inputs without AX permissions. If you or someone can educate me on this I'd be happy to make this change.

**Reply by ryanzigler** [reply]
> I don't know of a way to get global key inputs without AX permissions. If you or someone can educate me on this I'd be happy to make this change.

There are a couple of packages that manage to set global keyboard shortcuts without requiring AX permissions (or triggering any other permissions dialogs):
- [KeyboardShortcuts](https://github.com/sindresorhus/KeyboardShortcuts)
- [HotKey](https://github.com/soffes/HotKey)

I haven't looked into how exactly they are pulling this off if you wanted to implement it without a dependency, but I believe they are using the Carbon API for registering global keyboard shortcuts that hasn't been deprecated yet since Apple hasn't shipped a replacement API.

**Reply by mitchellh** [reply]
Thanks I’ll take a look assuming their license allows me to (haven’t checked yet).

**Reply by 00-kat** [reply]
Both are MIT.
